File: api/api.web.model/SearchWebModel.cs
using System;

namespace Apex.Api.Web.Model
{
    public class GamesQueryDto
    {
        public int? GameId { get; set; }
        public string DescriptionContains { get; set; }
        public string CreatedByUserName { get; set; }
        public string PlayerUserName { get; set; }
        public bool? ContainsMe { get; set; }
        public bool? IsPublic { get; set; }
        public bool? AllowGuests { get; set; }
        public GameStatusDto[] Statuses { get; set; }
        public DateTime? CreatedBefore { get; set; }
        public DateTime? CreatedAfter { get; set; }
        public DateTime? LastEventBefore { get; set; }
        public DateTime? LastEventAfter { get; set; }
    }

    public class SearchGameDto
    {
        public int Id { get; set; }
        public GameParametersDto Parameters { get; set; }
        public CreationSourceDto CreatedBy { get; set; }
        public GameStatusDto Status { get; set; }
        public DateTime LastEventOn { get; set; }
        public int PlayerCount { get; set; }
        public bool ContainsMe { get; set; }
        //TODO: Add current player name
    }
}

File: api/api.web/Model/SearchWebModel.fs
namespace Apex.Api.Web.Model

open System
open System.ComponentModel.DataAnnotations
open Apex.Api.Enums

[<CLIMutable>]
type GamesQueryDto = {
    gameId : Nullable<int>
    descriptionContains : string
    createdByUserName : string
    playerUserName : string
    containsMe : Nullable<bool>
    isPublic : Nullable<bool>
    allowGuests : Nullable<bool>
    statuses : List<GameStatus>
    createdBefore : Nullable<DateTime>
    createdAfter : Nullable<DateTime>
    lastEventBefore : Nullable<DateTime>
    lastEventAfter : Nullable<DateTime>
}

type SearchGameDto = {
    [<Required>]
    id : int

    [<Required>]
    parameters : GameParametersDto

    [<Required>]
    createdBy : CreationSourceDto
    
    [<Required>]
    status : GameStatus
    
    // Nullable
    lastEventOn : DateTime
    
    [<Required>]
    playerCount : int
    
    [<Required>]
    containsMe : bool
    //TODO: Add current player name
}